### ISSUE TITLE
Not able to send submission values by e-mail

### ISSUE THREAD

**Issue by maxiorel** [backdrop-contrib/webform#226]
Whenever I' trying to submit a webform with sending an e-mail enabled with submission values including [submission:values] or [submission:values:ANYTHING], I got error and blank screen with logo only:

ArgumentCountError: Too few arguments to function _webform_display_textfield(), 1 passed in /web/htdocs/www.XXXXX.cz/home/www/core/includes/theme.inc on line 1165 and at least 2 expected in function _webform_display_textfield() (line: 213 in file /web/htdocs/www.lkdenta.cz/home/www/modules/webform/components/textfield.inc).

Workaround is sending e-mails without [submission:values] token which is not good solution.

**Comment by olafgrabienski** [comment]
Interesting! On my site, I _am_ able to send submission values by e-mail. In my case, the values have the appendix `:nolabel`, for example: [submission:values:event:nolabel]

Are you using appendixes in your tokens? I'm not sure if they are (supposed to be) necessary, but maybe worth a try.

**Comment by maxiorel** [comment]
No luck, just tried and the same error with [submission:values:layout:left:your_name:nolabel]

**Comment by olafgrabienski** [comment]
Hm, what does the `layout:left` part mean? Is it a fieldset?

**Comment by maxiorel** [comment]
It is Layout box (layout) > Fieldset (left) > Field (your_name)

**Comment by olafgrabienski** [comment]
Ok, so one idea: Have you tried to use just `[submission:values:your_name:nolabel]`, maybe after moving the webform component outside of any wrapper?

**Comment by maxiorel** [comment]
:-) That was my first idea, but no luck.

Anyway, I think the base token [submission:values] should work in any cases. No need to specify each field.

**Comment by maxiorel** [comment]
Tried whole new webform without layouts and fields. Still the same problem. Also tried PHP 7.4, 8.0, 8.1, same error.

**Comment by maxiorel** [comment]
Quick and dirty fix:
webform.token.inc, line 168. This is problematic:
$replacements[$original] = backdrop_render($submission_renderable);

I've changed it to:
$data = array();
array_walk_recursive($submission->data, function($a) use (&$data) { $data[] = $a; });
$replacements[$original] = implode("\n", $data );

And now [submission:values] token works.

**Comment by olafgrabienski** [comment]
Glad to hear you have a fix! I'm still wondering why the submission values didn't work on your site in the first place. (As they work on my site.)

**Comment by argiepiano** [comment]
It seems to me that you have a faulty textfield component in your webform, which produces the error. Can you provide more details about the components?

**Comment by maxiorel** [comment]
> It seems to me that you have a faulty textfield component in your webform, which produces the error. Can you provide more details about the components?

I don't think so. As mentioned above, I created another webform with one textfield only and got the same error if default mailing enabled.

**Comment by argiepiano** [comment]
If this is a bug, the only way to fix it is to be able to reproduce it. The info you've provided is not enough to reproduce it. I'm trying on a clean install, with Webform with one textfield component, and the token replacement in the email is working fine. @olafgrabienski also has noticed it's working for him. The only explanation is that you may have something in your site that's interfering with the normal actions.

Do you have any other webform-related contrib installed? Do you have Entity Tokens installed? It may be a good idea to provide a list of contrib modules.

**Comment by argiepiano** [comment]
Also what do you mean by "default mailing enabled"?

**Comment by argiepiano** [comment]
Have you tried playing with the settings about empty components?
<img width="523" alt="Screen Shot 2023-08-17 at 1 57 06 PM" src="https://github.com/backdrop-contrib/webform/assets/9938978/c6f7d38f-2cb1-43d6-bce0-4f8d419119a3">

BTW, re-reading your tokens above - where did you get those "layout:left" tokens? Are those actually listed in the token modal that lists all available tokens? Those do not exist in my installation. Are those produced by a contrib module?

**Comment by maxiorel** [comment]
Sorry. I mean default e-mail template for webform submission. Forget my first real life example with layouts. To simplify the problem, I've created simple webform with one (required) textfield only and the problem is the same with default e-mail template. There are not any other modules working with webforms, forms at all or with tokens. The problem occurs in backdrop_render($submission_renderable) as I mentioned.

**Comment by yorkshire-pudding** [comment]
@maxiorel - is this the same as this issue: https://github.com/backdrop-contrib/webform/issues/105#issuecomment-1454715801 ?

**Comment by maxiorel** [comment]
@yorkshire-pudding Hi, I'm not sure, because I have tokens allowed. And it didn't help.
